tests/cloud_functions/token_refresh/test_main.py

```
@@ -0,0 +1,117 @@
+    mock_verify.return_value = mock_decoded_token
+
+    with patch("google.cloud.firestore.Client", return_value=db):
+        app = flask.Flask(__name__)
+        with app.app_context():
+            with app.test_request_context(
+                path="/",
+                method="POST",
+                headers={"Authorization": "Bearer test_token"},
+                query_string={"data_source": "strava"},
+                json={},
+            ):
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)